generators/cypress/generator.mts

```diff
@@ -27,24 +27,20 @@ import { CLIENT_MAIN_SRC_DIR } from '../generator-constants.mjs';
 import { GENERATOR_CYPRESS, GENERATOR_BOOTSTRAP_APPLICATION } from '../generator-list.mjs';
 
 import type { CypressApplication } from './types.mjs';
-import type {
-  LoadingTaskGroup,
-  PostWritingTaskGroup,
-  PreparingEachEntityTaskGroup,
-  PreparingTaskGroup,
-  PromptingTaskGroup,
-  WritingEntitiesTaskGroup,
-  WritingTaskGroup,
-} from '../base-application/tasks.mjs';
 import { generateTestEntity as entityWithFakeValues } from '../client/support/index.mjs';
+import { BaseApplicationGeneratorDefinition, Entity } from '../base-application/tasks.mjs';
 
 const { ANGULAR } = clientFrameworkTypes;
 
-/**
- * @class
- * @extends {BaseApplicationGenerator<CypressApplication>}
- */
-export default class CypressGenerator extends BaseApplicationGenerator<CypressApplication> {
+type ApplicationDefinition = {
+  applicationType: CypressApplication;
+  entityType: Entity;
+  sourceType: Record<string, (...args: any[]) => void>;
```

Comment:
Doc

Reply:
May this types extracted out of the generator.mts? In a dedicated file like `generator-types.d`?

Would be nice to slim down the generator LOCs

Reply:
IMO this is much better here.
It’s available at a glance where it’s used.

Reply:
I was taking my good old OOP ayatolla posture ^^. `type CypressApplicationDefinition extends ApplicationDefinition<CypressApplication>`

Reply:
I was taking my good old OOP ayatolla posture ^^. `type CypressApplicationDefinition extends ApplicationDefinition<CypressApplication>`